python/tests/io/test_pyarrow.py

```diff
@@ -0,0 +1,130 @@
+
+def test_pyarrow_input_file():
+    """Test reading a file using PyArrowInputFile"""
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        file_location = os.path.join(tmpdirname, "foo.txt")
+        with open(file_location, "wb") as f:
+            f.write(b"foo")
+
+        # Confirm that the file initially exists
+        assert os.path.exists(file_location)
+
+        # Instantiate the input file
+        absolute_file_location = os.path.abspath(file_location)
+        input_file = PyArrowInputFile(location=f"{absolute_file_location}")
+
+        # Test opening and reading the file
+        f = input_file.open()
+        data = f.read()
+        assert data == b"foo"
+        assert len(input_file) == 3
+
+
+def test_pyarrow_output_file():
+    """Test writing a file using PyArrowOutputFile"""
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        file_location = os.path.join(tmpdirname, "foo.txt")
+
+        # Instantiate the output file
+        absolute_file_location = os.path.abspath(file_location)
+        output_file = PyArrowOutputFile(location=f"{absolute_file_location}")
+
+        # Create the output file and write to it
+        f = output_file.create()
+        f.write(b"foo")
+
+        # Confirm that bytes were written
+        with open(file_location, "rb") as f:
+            assert f.read() == b"foo"
+
+        assert len(output_file) == 3
+
+
+def test_pyarrow_invalid_scheme():
+    """Test that a ValueError is raised if a location is provided with an invalid scheme"""
+
+    with pytest.raises(ValueError) as exc_info:
+        PyArrowInputFile("foo://bar/baz.txt")
+
+    assert ("PyArrowInputFile location must have a scheme of `file`, `mock`, `s3fs`, `hdfs`, or `viewfs`") in str(exc_info.value)
+
+    with pytest.raises(ValueError) as exc_info:
+        PyArrowOutputFile("foo://bar/baz.txt")
+
+    assert ("PyArrowOutputFile location must have a scheme of `file`, `mock`, `s3fs`, `hdfs`, or `viewfs`") in str(exc_info.value)
+
+
+@patch("iceberg.io.pyarrow.FileSystem")
+def test_pyarrow_violating_InputStream_protocol(MockedFileSystem):
+    """Test that a TypeError is raised if an input file is provided that violates the InputStream protocol"""
+
+    # Missing seek, tell, closed, and close
+    input_file_mock = MagicMock(spec=["read"])
+
+    # Create a mocked filesystem that returns input_file_mock
+    filesystem_mock = MagicMock()
+    filesystem_mock.open_input_file.return_value = input_file_mock
+
+    MockedFileSystem.from_uri.return_value = (
+        filesystem_mock,
+        "foo_path",
+    )  # Patch the FileSystem.from_uri method to return the mocked filesystem
+
+    with pytest.raises(TypeError) as exc_info:
+        PyArrowInputFile("foo.txt").open()
+
+    assert ("Object returned from PyArrowInputFile.open does not match the InputStream protocol.") in str(exc_info.value)
+
+
+@patch("iceberg.io.pyarrow.FileSystem")
+def test_pyarrow_violating_OutputStream_protocol(MockedFileSystem):
```

Comment:
I see this pattern used elswhere but mixing camel cass and snake case doesn't look write to me.

Reply:
Yeah it does look odd, I'll update this.